File: backend/app/schemas/assistant.py
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field

from app.schemas.common import BaseSchema


class AssistantChatRequest(BaseModel):
    """Payload to query the Agricultural AI Assistant."""

    message: str = Field(min_length=2, max_length=1000, description="Farmer query in English or Tamil")
    farm_id: Optional[str] = None
    zone_id: Optional[str] = None
    language: str = Field(default="en", description="en (English) or ta (Tamil)")


class AssistantChatResponse(BaseSchema):
    """Grounded, factual response synthesized from verified backend telemetry."""

    response_text: str
    language: str
    tools_used: List[str]
    data_sources: List[Dict[str, Any]]
    disclaimer: str = "Assistant responses are synthesized from verified PostGIS and AI model telemetry without hallucination."


class AssistantToolInfo(BaseSchema):
    """Metadata for registered internal query tools."""

    name: str
    description: str
    parameters: List[str]

File: frontend/src/types/assistant.ts
export interface AssistantChatRequest {
  message: string;
  farm_id?: string;
  zone_id?: string;
  language?: 'en' | 'ta' | 'hi' | 'mr' | string;
  image_base64?: string;
  image_mime?: string;
}

export interface AssistantChatResponse {
  response_text: string;
  language: string;
  tools_used: string[];
  data_sources: Record<string, any>[];
  disclaimer: string;
}

export interface AssistantToolInfo {
  name: string;
  description: string;
  parameters: string[];
}

export interface ChatMessage {
  id: string;
  sender: 'user' | 'assistant';
  text: string;
  image_url?: string;
  tools_used?: string[];
  timestamp: string;
  language?: string;
}

